app/controllers/codelists-management/index.js

```diff
@@ -30,27 +29,28 @@ export default class CodelistsManagementIndexController extends Controller {
     this.page = 0;
   }
 
-  @action openRemoveModal(codelist) {
-    this.modalCodelist = codelist;
-    this.isRemoveModalOpen = true;
-  }
-
-  @action closeRemoveModal() {
-    this.modalCodelist = null;
-    this.isRemoveModalOpen = false;
+  @action async removeCodelist(codelist) {
```

Comment:
This doesn't really need to be a task: showing the status (like loading) is not needed and this can't really be triggered twice. However, I guess it would be safer to encapsulate this in a droppedTask.